### DISCUSSION TITLE
Why doesn't Debian include your repos?

### DISCUSSION THREAD

**Discussion by stevespages** [oerdnj/deb.sury.org#1771] [General]
Hi,
I am an amateur web developer trying to get better and I have in the past (and am hoping to do so very soon) updated my PHP version on my Linux computer (normally Mint) using your repo which I am very grateful for.

What I do not understand is why, given that your up to date PHP versions are available, the latest stable version of Linux Mint which I have just installed (20.3 Cinnamon) comes with an old version of PHP (7.4) even after I run apt update. I imagine this is down to Debian.

As far as I can see this is a problem which applies to many of the programs on the computer. I have had to install appimages for nvim and new versions of node and npm to get a better working environment.

From my position of considerable lack of knowledge about the subject,  I have begun to wonder if it would be better if Linux distros stopped shipping with programs beyond the kernel and the desktop experience and left it to the user to get up to date software from the various sources such as yourself.

I would be very grateful if you could let me know what you think or point me to any articles or discussions about this issue which might help me understand what is going on.

Part of my concern from a personal point of view is that I think web development becomes very much more difficult if the developer has to configure and assemble their environment on their development computer and on their server as well as write all the code for the web site. I feel I am drowning in technical complexity which a good OS would handle for me if it was shipped with up to date software.

Steve

**Comment by r0mdau** [comment]
Hello @stevespages ,
Debian focuses on stability, this is why by design you will not run the latest versions. 

There are 3 [release cycles](https://www.debian.org/releases/) : 
* stable
* testing
* unstable

For production servers it is recommended running `stable` version and of course you will not have latest versions of packages. On my laptop I run `testing` which is stable enough, it depends, you have mostly not latest packages. Latest packages are on the `unstable` release.

It is known to use Ubuntu non-LTS (Long Term Support) versions as a developer on your workstation to get fresher package versions.

Arch Linux is well-known to iterate quickly and integrate latest packages versions.

The last option is to add custom APT repositories from the developper to your Debian. Like with Sury repo.
